kore/src/Kore/Step/Axiom/Identifier.hs

```diff
@@ -85,24 +91,25 @@ Returns 'Nothing' if the 'TermLike' does not conform to one of the structures we
 recognize.
 
  -}
-matchAxiomIdentifier :: TermLike variable -> Maybe AxiomIdentifier
-matchAxiomIdentifier = Recursive.fold matchWorker
+matchAxiomIdentifier
+    :: InternalVariable variable
+    => TermLike variable
+    -> Maybe AxiomIdentifier
+matchAxiomIdentifier = Recursive.fold matchWorker . Builtin.externalize
```

Comment:
Is `externalize` evaluated lazily?

Reply:
Yes, it is defined by `Recursive.unfold`, so it lazily unfolds the `TermLike`.